### PR TITLE
JATaber welcome message and verify

### PR DESCRIPTION
**Pull request by JATaber**

### CODE REVIEW

**Comment by mpeck99** [comment]
Looks good!

**Comment by JATaber** [comment]
Thank you very much!

File: bot/client.js [hunk 1]
```diff
@@ -15,6 +15,14 @@ client.on('ready', () => {
   util.log('Bot Online and Ready', 0);
 });
 
+// Message for new users
+client.on('guildMemberAdd', (member) => {
+  // Send Direct Message with Terms of Service
+  member.send(`Welcome ${member.user.username} to the family! Please read our Terms of Service`);
+  // Send Welcome User Message in the General Channel
+  member.guild.channels.find('name', 'general').send(`Welcome ${member.user.username} to the Family!`);
```
Comment (reactivepixel) [inline bot/client.js:23]: When we bury strings inside of code, even somewhat non-complex code, we make it hard to maintain the text messages we are trying to send. This is strongly related to [magic numbers](https://codeburst.io/software-anti-patterns-magic-numbers-7bc484f40544) in that if we implement a variable, and control that string within, using the variable on the sending of the message method will come out cleaner. If you look at the email being sent in the verify controller you can see how quickly inline strings used in this manner begin to degrade.

```javascript
const targetChannel = 'general';
const greeting = 'Welcome ${member.user.username} to the Family!';

// Send Welcome User Message in the General Channel
member.guild.channels.find('name', targetChannel).send(greeting);
```

File: bot/controllers/tosWelcome.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    super(message);
+    // Aliasing 'this' as controller to allow for binding in actions
+    const controller = this;
+    // Array of all commands, see baseCommand.js for prototype
+    this.commands = [
+      new Command(
+        '!tos',
+        '!tos',
+        'Resend Terms',
+        'Send Terms Copy',
```
Comment (reactivepixel) [inline bot/controllers/tosWelcome.js:16]: Even though we are short handing the the ToS down to a simple one liner, the help command should at least specify that it's the Terms of Service. 'Resend Terms' and 'Send Terms copy' would be acceptable for our internal use, but consider that this help message is user facing... and [users will surprise you with how they interact with your products](https://twitter.com/designuxui/status/576432203560685568?lang=en).

File: bot/client.js [hunk 3]
```diff
@@ -15,6 +15,14 @@ client.on('ready', () => {
   util.log('Bot Online and Ready', 0);
 });
 
+// Message for new users
+client.on('guildMemberAdd', (member) => {
+  // Send Direct Message with Terms of Service
+  member.send(`Welcome ${member.user.username} to the family! Please read our Terms of Service`);
```
Comment (reactivepixel) [inline bot/client.js:21]: Duplicate code logic from within your controller.

File: bot/controllers/tosWelcome.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        this.welcomeAction.bind(controller),
+        'dm',
+      ),
+    ];
+  }
+  // Sends the welcome message to the user
+  welcomeAction() {
+    const { message } = this;
+    return `Welcome ${message.author.username} to the server! Please read our Terms of Service. If you want to receive the terms again just type !tos`;
```
Comment (reactivepixel) [inline bot/controllers/tosWelcome.js:25]: The user has entered ```!tos``` to execute this controller, providing them with instructions to execute this controller, at this point, is redundant.

**Comment by eMediaLab** [comment]
A more explicit PR title will help others understand the context for this work.

**Review by reactivepixel — CHANGES_REQUESTED** [review]
Very good overall James! Only a few touches I would like to have seen cleaned up.

Additionally, the duplicate code logic I mentioned was the real trick of the assignment. Your solution was an okay one, standard even, but when building custom systems we want to minimize all duplication of code. As you can imagine this small duplication shouldn't be too bad to maintain... but over the months and years of the life cycle of this codebase, with many hands adding code, and each hand adding some non-DRY piece like this, could really conflate the codebase and drift toward a patchwork quilt of one offs and redundancies.

**Comment by reactivepixel** [comment]
A few additional notes @JATaber,

"Finally Finished" is an inappropriate pull request name. Additionally, your commit message of "hopefully this goes to the correct spot this time" is also inappropriate.

Additionally you only have a single commit made in this PR. By only having a single commit, you really open yourself to potential risk of losing your work due to a system crash. When working for a client, and your being payed for your time, they will typically want a return on that. Consider that if at the end of a long week of coding you're about ready to commit your changes and push them to the client's repo and your computer's hard drive melts down losing your week's worth of work... should the client still pay you for that time you put in? 

Committing frequently has many advantages, while single commits are fraught with disadvantages. This is a habit you need to immediately abandon moving forward.

**Pull request closed without merging**